internal/command/limits.go

```diff
@@ -0,0 +1,106 @@
+package command
+
+import (
+	"context"
+	"time"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/command/preparation"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/errors"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/limits"
+)
+
+type SetLimits struct {
+	AuditLogRetention *time.Duration `json:"AuditLogRetention,omitempty"`
+}
+
+// SetLimits creates new limits or updates existing limits.
+func (c *Commands) SetLimits(
+	ctx context.Context,
+	resourceOwner string,
+	setLimits *SetLimits,
+) (*domain.ObjectDetails, error) {
+	instanceId := authz.GetInstance(ctx).InstanceID()
+	wm, err := c.getLimitsWriteModel(ctx, instanceId, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+	aggregateId := wm.AggregateID
```

Comment:
Is it intended that limits is a separate aggregate? As it is still linked to an instance

Reply:
You mean a separate aggregate type? Yes, it's intended.
We don't pollute the already big instance aggregate unnecessarily. The limits don't need to be in the instances transaction boundery, IMO.
Also, it gives us a lot of flexibility in the future, see PR description.

Reply:
Or do you mean that we can reuse the instance id as the aggregate id?

Reply:
Thought as much, to not pollute the instance aggregate, would still generate a separate ID as the instanceID is contained in the data.